rz_sys_system should not depend on LIBC_HAVE_FORK

ret2libc opened the pull request:
This PR should fix current failures in AppVeyor. In particular, it moves all the arch-dependent code to sys.c and it implements `rz_sys_system` in different ways based on the availability of the host machine. Moreover it changes the signature of `rz_sandbox_system` so that it always does a `system`. In some cases `rz_sandbox_system` was called to actually execute just a `/bin/sh -c` in the current process, but I changed those few callers to just call `rz_sys_execl`, because I think "system" has a "standard meaning" which is `fork+exec(/bin/sh -c)`.

ret2libc committed:
Revert "Use async-signal-safe functions after fork in multithread program (#207)"

This reverts commit a0e880899491f1b94859bea253cb4415abb0f13b.

ret2libc commented:
Somehow it's this commit that broke dev, though it's not clear to me why.

ret2libc committed:
rz_sys_system should not depend on LIBC_HAVE_FORK

ret2libc committed:
Revert "Revert "Use async-signal-safe functions after fork in multithread program (#207)""

This reverts commit b6cb7a8ec4561c89247e0c710db132dbbac06b00.

ret2libc commented on librz/util/sandbox.c:
```diff
@@ -210,7 +210,6 @@ RZ_API int rz_sandbox_system(const char *x, int n) {
 		return -1;
 	}
-#if LIBC_HAVE_FORK
 #if LIBC_HAVE_SYSTEM
```
maybe all this thing should be inside `rz_sys_system`.

ret2libc commented:
@XVilka AppVeyor still not running on PR :(

kazarmy committed:
Merge branch 'dev' into fix-appveyor

kazarmy commented on librz/util/sandbox.c:
```diff
@@ -210,7 +210,6 @@ RZ_API int rz_sandbox_system(const char *x, int n) {
-#if LIBC_HAVE_FORK
 #if LIBC_HAVE_SYSTEM
 	if (n) {
 #if APPLE_SDK_IPHONEOS
```
```suggestion
#if APPLE_SDK_IPHONEOS && LIBC_HAVE_FORK
```

ret2libc replied:
I've actually changed the whole thing so that `rz_sys_system` implements the right thing depending on the availability of `system`/`fork` on the host machine. So now `rz_sandbox_system` just checks if sandboxing is enabled and then it calls `rz_sys_system`.

kazarmy replied:
Oh ok

ret2libc committed:
Make rz_sandbox_system just call rz_sys_system

To improve consistency, "system" calls "system" syscalls when available
or it emulates it. If one does not want the fork associated with system,
he should just use rz_sys_exec* functions.

ret2libc committed:
Try changed version for appveyor

ret2libc committed:
Merge branch 'dev' into fix-appveyor
